SofaKernel/modules/SofaCore/src/sofa/core/objectmodel/BaseData.h

```diff
@@ -171,7 +171,7 @@ class SOFA_CORE_API BaseData : public DDGNode
     void setWidget(const char* val) { widget = val; }
 
     /// True if the counter of modification gives valid information.
-    SOFA_DEPRECATE_BASEDATA_API("This method has been removed as Data<> must have, by design, their counter valid.")
+    SOFA_ATTRIBUTE_DEPRECATED("2020-10-03 PR#1503", "v21.12", "Data<> must have, by design, their counter valid.")
```

Comment:
There should be a macro SOFA_ATTRIBUTE_DEPRECATE_BASEDATA_API instead of using a generic macro. To me macro name are used to indicate a groupe of related changes that have the same deprecation timeline. The use of generic macro breaks that (in addition to duplicate all the dates)